TimeFormat.java

```diff
@@ -1,17 +1,29 @@
 // Represents the hh:mm time format using an AM/PM format. 
 public class TimeFormat {
 	public static void main(String[] args) {
-		// In Java, the command-line arguments args[0], args[1], ... are represented
-		// each by a string. In this program, the single "hh:mm" input is represented
-		// by the single command-line string argument args[0]. 
-		//   
-		// The following statement handles the hours part of the input.
-		// It concatenates the empty string "" with the leftmost hour-digit. 
-		// It then concatenates the resulting string with the rightmost hour-digit,
-		// and then uses parseInt to cast the resulting string as an int.
 		int hours = Integer.parseInt("" + args[0].charAt(0) + args[0].charAt(1));
-		// Does the same with the minutes part of the input.
 		int minutes = Integer.parseInt("" + args[0].charAt(3) + args[0].charAt(4));
-        // Replace this comment with the rest of your code
+		String pmOrAmStr="AM";
+		if (hours<10) {
+			hours = hours%10;
+		}
+		if (hours>12) {
+			pmOrAmStr = "PM";
+			hours = hours-12;
+		}
+		if (hours==12) {
+			pmOrAmStr = "PM";
+		}
+		if (minutes<10) {
+			if (minutes>=1) {

```

Comment:
1. Spacing
2. Bad indentations
3. redundant split.. 
I am talking about the inner conditions `(if (minutes>=1))` and `if (minutes<1)`

First, there is a difference between 2 ifs and if else. In this case the second fits since if minutes is not bigger or equal to 1 it is smaller.

Second, 
Since `minutes` can be only non negative number (by definition). 
Hence, if the number in minutes is entering the outer if's condition `(minutes < 10)`. 
It means that means that the cases entering this place are `0,1,2,...,9`.

Lets go for the 2nd inner condition for a second.`if (minutes<1)`
If the 2nd inner condition holds and enter the condition, and since minutes is in range [0,9]. 
It means that `minutes == 0` always. hence: 
Writing the statement `"00".equals("0" + minutes)` always holds, **in the second inner if**
`Reminder: "" + 0 = "0" (String + int will result in String)`
Therefore I could have wrote under the second condition
`System.out.println(hours + ":" + "0" + minutes + " " + pmOrAmStr);`
Which is the same statement under the first inner if. 
And if I have always do an operation I don't need to have condition. 
example: 
```
if (agreed) {
   runTest(user);
} else {
   runTest(user);
}
```
We can write just `runTest(user);`